docs/admin/git-providers.md

```diff
@@ -0,0 +1,80 @@
+# Git Providers
+
+Coder integrates with git providers so developers can authenticate with repositories within their workspace.
+
+## How it works
+
+When developers use `git` inside their workspace, they are first prompted to authenticate. After that, Coder will store/refresh tokens for future operations. 
+
+<video autoplay playsinline loop>
+  <source src="https://github.com/coder/coder/blob/main/site/static/gitauth.mp4?raw=true" type="video/mp4">
+Your browser does not support the video tag.
+</video>
+
+## Configuration
+
+To add a git provider, you'll need to create an OAuth application. The following providers are supported:
+
+- [GitHub](https://docs.github.com/en/developers/apps/building-oauth-apps/creating-an-oauth-app) (GitHub apps are also supported)
+- [GitLab](https://docs.gitlab.com/ee/integration/oauth_provider.html)
+- [BitBucket](https://support.atlassian.com/bitbucket-cloud/docs/use-oauth-on-bitbucket-cloud/)
+- [Azure DevOps](https://learn.microsoft.com/en-us/azure/devops/integrate/get-started/authentication/oauth?view=azure-devops)
+
+Example callback URL: `https://coder.example.com/api/v2/gitauth/primary-github/callback`. Use an arbitrary ID for your provider (e.g. `primary-github`).
```

Comment:
@kylecarbs what URL do you use if an ID is not set? I was a bit confused

Reply:
It'll be `/gitauth/<id but falls back to type>/callback`

Reply:
btw @bpmct it's not `/api/v2`. I moved the callback to the root for a nicer route.